compiler/rustc_target/src/spec/apple_base.rs

```diff
@@ -274,6 +320,11 @@ fn ios_deployment_target() -> (u32, u32) {
     from_set_deployment_target("IPHONEOS_DEPLOYMENT_TARGET").unwrap_or((7, 0))
 }
 
+fn mac_catalyst_deployment_target() -> (u32, u32) {
+    // If you are looking for the default deployment target, prefer `rustc --print deployment-target`.
+    from_set_deployment_target("IPHONEOS_DEPLOYMENT_TARGET").unwrap_or((14, 0))
```

Comment:
I pulled it from the original implementation of this behavior that only applied to catalyst https://github.com/rust-lang/rust/pull/114114/files#diff-6bbf6edd930f26d3b180eb5c250b0e48d8f3c5eb474e3274909ef6ae6f0d1e61L343

I agree having it configurable makes sense, at least in this case if you set `IPHONEOS_DEPLOYMENT_TARGET` it would win though

Reply:
I missed the lines removed in the diff, oops. That answers the immediate question then.

> I agree having it configurable makes sense, at least in this case if you set IPHONEOS_DEPLOYMENT_TARGET it would win though

Do you want to take that to a followup PR or should I? Fixing the mess that's Catalyst has been on my list forever.

Reply:
I can send that one after!

Reply:
ah it looks like using the triple for this case https://github.com/rust-lang/rust/commit/78bbe57c88469abbbcd1c46c8e8f3cf014b5a154#diff-f2aa7189f140501f7d0d283ef503e1980a9449ef0292b48b650f6e0c87bb8d6b complicates things slightly

Reply:
iiuc it should just be making another of the `_lld_platform_version` functions to use for this and other LLVM needs and have that source from the deployment target function. Might have missed a complication with the catalyst target though.

Reply:
Ok sounds good I can try it after this one

Reply:
It seems like to make this work I would need to change `add_pre_link_args` to take a `String` or something else since ours would be dynamically computed here, would that be a reasonable change?